### ISSUE TITLE
github layer: magit keys missing from the magit help

### ISSUE THREAD

**Issue by tko** [syl20bnr/spacemacs#2086]
github layer adds a few keybindings in magit-status-mode. However, those keys are not behaving like magit keys otherwise; they don't show up in the list of keys/commands you see with `?`, when you hit the new `#` command/prefix key you still don't get any further guidance, neither from magit nor from guide-key.

**Comment by tko** [comment]
Something along the lines of
```elisp
(magit-key-mode-insert-action 'dispatch (kbd "#") "GitHub-PR" 'magit-key-mode-popup-github)
```
might work (ref: terranpro/magit-gerrit#17) .. might need to implement the popup as well, I guess.

**Comment by TheBB** [comment]
The proposed code doesn't work in the new version of Magit, but you can use `magit-define-popup-key` and `magit-change-popup-key` to get the same effect.

Unfortunately it's not so clear how to do this in a way that works with both evilified and emacs state. Simplest would be to go with one of them on startup and say that if you switch while Spacemacs is running, that's your own problem.

**Comment by justbur** [comment]
Fixed with #3578

**Comment by tarsius** [comment]
> those keys [...] don't show up in the list of keys/commands you see with `?`

Since this doesn't affect only Spacemacs, the respective bindings should be established in the upstream packages.

For `magit-gh-pulls` I have done so at sigma/magit-gh-pulls#96. Someone(tm) please do that for the other extensions. See that pr for how to do it.
